### DISCUSSION TITLE
Cannot read properties of null (reading 'appendChild')

### DISCUSSION THREAD

**Discussion by fampeeters** [CodingTrain/website#3338] [Q&A]
The videos about p5.js on TheCodingTrain are fantastic! I now downloaded a course p5.js in Dutch for my students and in every example of that course there is a line in the function setup():
canvas.parent('processing');

I believe the line is meant to create a div in the html.
However, I get an error message every time I use this line:
**TypeError: Cannot read properties of null (reading 'appendChild')**
I just comment that line so that it is ignored. But can I solve the problem in a different way so that I can keep all the examples with this line included?

```
function setup() {
  canvas = createCanvas(450,450);
  background('darkred');
  // canvas.parent('processing');
  noLoop();
}

```

**Comment by holasoymalva** [comment]
Of course, the only thing necessary is to add a `div` in the html file with the `id="processing"` 🙂 You can take the following example as a reference: 👉 [CLICK HERE](https://codepen.io/malvabombom/pen/qBVgZWX), or if would you like to create the div inside the JS file with P5 you can use this example as a reference 👉 [CLICK HERE](https://editor.p5js.org/alchemist3301/sketches/C0w_BAfOo)  In these cases the use of **_.parent()_** is to attach the element to the parent specified (That Parent should exist before attaches). You can read all the documentation about this method in the next link 👉 [Read HERE](https://p5js.org/es/reference/#/p5.Element/parent), My best wishes ✨
